plotly/plotly/plotly.py

```diff
@@ -1308,7 +1308,19 @@ def response_handler(cls, response):
                     .format(url=get_config()['plotly_api_domain'])
                 )
             else:
-                raise requests_exception
+                try:
+                    parsed_response = response.json()
+                except:
+                    parsed_response = response.content
+
+                try:
+                    # try get a friendly error message
+                    errors = parsed_response.get('errors')
+                    message = errors[-1]['message']
+                except:
+                    raise requests_exception
+
+                raise exceptions.PlotlyError(message)
```

Comment:
I feel like we *just* did this:

https://github.com/plotly/plotly.py/blob/master/plotly/plotly/plotly.py#L1704-L1720

Want to try and follow that more closely, OR, extract the common functionality out? It'd be great to just have one error handler.

In particular, the code you wrote assume that `errors` is *not empty*, which we might not want to assume.